bookkeeper-common/src/main/java/org/apache/bookkeeper/common/net/ServiceURI.java

```diff
@@ -215,21 +249,21 @@ public static ServiceURI create(URI uri) {
     }
 
     private static String validateHostName(String serviceName, String hostname) {
-        String[] parts = hostname.split(":");
-        if (parts.length >= 3) {
+        URI uri = null;
+        try {
+            uri = URI.create("dummyscheme://" + hostname);
```

Comment:
`uri = URI.create("dummyscheme://" + hostname);`
I think testcase doesn't cover this

Reply:
In fact any one of testcase in ServiceURITest could cover this,  `ServiceURI#create` call `ServiceURI#validateHostName` to validate the format of host name.